src/react-hooks-nesting-walker/is-hook-identifier.ts

```diff
@@ -1,6 +1,6 @@
 import { Identifier } from 'typescript';
 
-export function isHookIdentifier(node: Identifier) {
+export function isHookIdentifier(node: Identifier): node is Identifier {
```

Comment:
`node` is already typed as an `Identifier`, so it seems to me that the return type is not necessary 🙂

Reply:
Since it has been some time that I created this PR I don't remember all the details, but I remember that I needed this [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) in some place to be able to access it's properties, since the input type was actually not Identifier but something more general... in case we should decide to land this PR I will give more details at some point in time.